components/purchase_modal/purchase_modal.tsx

```diff
@@ -59,31 +63,51 @@ type State = {
     billingDetails: BillingDetails | null;
     cardInputComplete: boolean;
     processing: boolean;
-    selectedProduct: Product | null | undefined;
+    editPaymentInfo: boolean;
+    currentProduct: Product | null | undefined;
+    nextProductToUpgrade: Product | null | undefined;
```

Comment:
I believe it was more clear before as `selectedProduct` given that they may be able to downgrade from where they start the trial in Enterprise and downgrade to Starter for example.

```suggestion
    selectedProduct: Product | null | undefined;
```